sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -353,6 +367,30 @@ def create_or_update(
         name, version = component._get_rest_name_version()
         rest_component_resource = component._to_rest_object()
         result = None
+        try:
+            if not component._is_anonymous:
+                client_component_hash = component.get_component_hash(keys_to_omit=["version"])
+                rest_component_resource.properties.properties["client_component_hash"] = client_component_hash
+                remote_component_version = self.get_component_version(
+                    name=name
+                )  # will raise error if not found, so need set client_component_hash in the above line first.
+                remote_component_hash = None
+                if remote_component_version:
+                    remote_component_hash = remote_component_version.properties.properties.get("client_component_hash")
+                if remote_component_hash == client_component_hash and kwargs.get("skip_if_no_change"):
+                    component.version = remote_component_version.properties.component_spec.get(
+                        "version"
+                    )  # only update the default version component instead of creating a new version component
+                    version = component.version
+                    rest_component_resource = component._to_rest_object()
```

Comment:
What is the difference between this and the created one in line 368?

Reply:
Because the component version is updated, and I assume I don't know whether the `ComponentVersion `object constructed by the `to_rest_object() `function will be different due to the change of version, so I will reconstruct it again.